exporter/opensearchexporter/internal/objmodel/objmodel.go

```diff
@@ -0,0 +1,535 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+// The objmodel package provides tools for converting OpenTelemetry Log records into
+// JSON documents.
+//
+// The JSON parsing in OpenSearch does not support parsing JSON documents
+// with duplicate fields. The fields in the docuemt can be sort and duplicate entries
+// can be removed before serializing. Deduplication ensures that ambigious
+// events can still be indexed.
+//
+// With attributes map encoded as a list of key value
+// pairs, we might find some structured loggers that create log records with
+// duplicate fields. Although the AttributeMap wrapper tries to give a
+// dictionary like view into the list, it is not 'complete'. When iterating the map
+// for encoding, we still will encounter the duplicates.
+// The AttributeMap helpers treat the first occurrence as the actual field.
+// For high-performance structured loggers (e.g. zap) the AttributeMap
+// semantics are not necessarily correct. Most often the last occurrence will be
+// what we want to export, as the last occurrence represents the last overwrite
+// within a context/dictionary (the leaf-logger its context).
+// Some Loggers might even allow users to create a mix of dotted and dedotted fields.
+// The Document type also tries to combine these into a proper structure, such that these mixed
+// representations have a unique encoding only, which allows us to properly remove duplicates.
+//
+// The `.` is special to OpenSearch. In order to handle common prefixes and attributes
+// being a mix of key value pairs with dots and complex objects, we flatten the document first
+// before we deduplicate. Final dedotting is optional and only required when
+// Ingest Node is used. But either way, we try to present only well formed
+// document to OpenSearch.
+
+// nolint:errcheck
```

Comment:
Can this be scoped to specific statements?  Even if that requires duplication it's preferable to not allowing further unchecked errors to be introduced without discussion.